source/code/source/playfab/PlayFabEventPipeline.cpp

```diff
@@ -272,10 +272,10 @@ namespace PlayFab
                 // call individual emit event callbacks
                 for (const auto& eventEmitRequest : *requestBatchPtr)
                 {
-                    const auto& playFabEmitRequest = std::dynamic_pointer_cast<const PlayFabEmitEventRequest>(eventEmitRequest);
+                    std::shared_ptr<const PlayFabEmitEventRequest> playFabEmitRequest = std::dynamic_pointer_cast<const PlayFabEmitEventRequest>(eventEmitRequest);
                     auto playFabEmitEventResponse = std::shared_ptr<PlayFabEmitEventResponse>(new PlayFabEmitEventResponse());
                     playFabEmitEventResponse->emitEventResult = EmitEventResult::Success;
-                    auto playFabError = std::shared_ptr<PlayFabError>(new PlayFabError());
+                    std::shared_ptr<PlayFabError> playFabError = std::shared_ptr<PlayFabError>(new PlayFabError());
```

Comment:
>std::shared_ptr<PlayFabError> [](start = 20, length = 29)

you've expanded this type but the `playFabEmitEventResponse` variable 2 lines above which uses the same initialization pattern didn't change. seems inconsistent.

frankly this is an instance where I'd just use auto + make_shared (as alluded to in another comment)

`auto playFabError = std::make_shared<PlayFabError>();`

or if you really want to expand the type you can do...

`std::shared_ptr<PlayFabError> playFabError (new PlayFabError());`

this latter one eliminates the need to put std::shared_ptr on both sides which makes the line easier to read

Reply:
from your description it sounds like either of these options would more align with the auto rules you are trying to establish. Either using auto because the type is obviously on the right hand side or avoiding the left vs right hand type question entirely

---
In reply to: [340922563](https://github.com/PlayFab/XPlatCoreTemplate/pull/33#discussion_r340922563) [](ancestors = 340922563)